### ISSUE TITLE
Angular 'Http' deprecated in favour of 'HttpClient'

### ISSUE THREAD

**Issue by subotic** [dhlab-basel/Salsah#156]
The Angular `Http` from `@angular/http' was deprecated. We should use `HttpClient` from `@angular/common/http`.

For usage see https://angular.io/guide/http. This also affects mocking of HTTP requests, which would need to be refactored to use the `HttpClientTestingModule` (see: https://angular.io/guide/http#testing-http-requests)

**Comment by subotic** [comment]
@kilchenmann do you want to work on this or should I? This is kinda very important because there is a chance that our code won't work with the next version of Angular.

**Comment by kilchenmann** [comment]
Yes, this is very important indeed and I wanted to work on that, but right now I don't have the time. So, feel free to work on it, if you like.
